cmd/run.go

```diff
@@ -4,77 +4,119 @@ import (
-{{rootCmdUse}} run --builder s2i
-
-# Run the function, forcing a rebuild of the image.
-#   This is useful when the function's image was manually deleted, necessitating
-#   A rebuild even when no changes have been made the function's source.
-{{rootCmdUse}} run --build
-
-# Run the function, forcing a rebuild of the image with --builder option.
-{{rootCmdUse}} run --build --builder s2i
-
-# Run the function's existing image, disabling auto-build.
-#   This is useful when filesystem changes have been made, but one wishes to
-#   run the previously built image without rebuilding.
-{{rootCmdUse}} run --build=false
-
+		Long: `
+NAME
+	{{rootCmdUse}} run - Run a function locally
+
+SYNOPSIS
+	{{rootCmdUse}} run [-t|--container] [-r|--registry] [-i|--image] [-e|--env]
+	             [--build] [-b|--builder] [--builder-image] [-c|--confirm]
+	             [-v|--verbose]
+
+DESCRIPTION
+	Run the function locally.
+
+	Values provided for flags are not persisted to the function's metadata.
+
+	Containerized Runs
+	  The --container flag indicates that the function's container shuould be
+	  run rather than running the source code directly.  This may require that
+	  the function's container first be rebuilt.  Building the container on or
+	  off can be altered using the --build flag.  The default value --build=auto
+	  indicates the system should automatically build the container only if
+	  necessary.
+
+	Process Scaffolding
+	  This is an Experimental Feature currently available only to Go projects.
+	  When running a function with --container=false (host-based runs), the
+	  function is first wrapped code which presents it as a process.
+	  This "scaffolding" is transient, written for each build or run, and should
+	  in most cases be transparent to a function author.  However, to customize,
+	  or even completely replace this scafolding code, see the 'scaffold'
+	  subcommand.
+
+EXAMPLES
+
+	o Run the function locally from within its container.
+	  $ {{rootCmdUse}} run
+
+	o Run the function locally from within its container, forcing a rebuild
+	  of the container even if no filesysem changes are detected
+	  $ {{rootCmdUse}} run --build
+
+	o Run the function locally on the host with no containerization (Go only).
+	  $ {{rootCmdUse}} run --container=false
 `,
 		SuggestFor: []string{"rnu"},
-		PreRunE:    bindEnv("build", "path", "builder", "registry", "verbose"),
+		PreRunE:    bindEnv("build", "builder", "builder-image", "confirm", "env", "image", "registry", "path", "container", "verbose"),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			return runRun(cmd, args, newClient)
 		},
 	}
 
+	// Global Config
 	cfg, err := config.NewDefault()
 	if err != nil {
 		fmt.Fprintf(cmd.OutOrStdout(), "error loading config at '%v'. %v\n", config.File(), err)
 	}
 
+	// Function Context
+	f, _ := fn.NewFunction(effectivePath())
+	if f.Initialized() {
+		cfg = cfg.Apply(f)
+	}
```

Comment:
If the function isn't initialized, it can't be run, right? Is that handled elsewhere?

Reply:
Yes that's handled elsewhere.  I could refactor this a bit to make it more clear, but for right now all of the commands which use "the current function" to override global values only do so if there is actually an initialized functoin at that path (otherwise, as you say, the error of "can't X; no function at Y" is handled later.  This stanza is the pre-execution phase where flag defaults are being populated, so it is a bit more forgiving.    Definitely could refactor this if statement away entirely, however.